homeassistant/components/switchbot/__init__.py

```diff
@@ -43,6 +42,21 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+class OptionsListener:
+    """Listen for changes in options."""
+
+    def __init__(self, original_options: Mapping[str, Any]) -> None:
+        """Initialize the options listener."""
+        self._original_options = dict(original_options)
+
+    async def async_update_listener(
+        self, hass: HomeAssistant, entry: ConfigEntry
+    ) -> None:
+        """Handle options update."""
+        if dict(entry.options) != self._original_options:
```

Comment:
Why do we need this check? Is this listener called sometime when the options aren't changed?

Reply:
You are right, we don't need to check as the bluetooth address will never change.  Force of habit for DHCP changes 😄 

Will drop the check